doc/developer-guides/hld/hld-devicemodel.rst

```diff
@@ -789,6 +789,137 @@ the bus hierarchy would be:
+------------------------
+
+To simplify PTM support implementation, the virtual root port only supports the
+most basic PCIe configuration and operation, in addition to PTM capabilities.
+
+In Guest VM post-launched scenarios, you enable PTM by setting the
+``enable_ptm`` option for the pass through device (as shown above).
+
+.. figure:: images/PTM-hld-PTM-flow.png
+   :align: center
+   :width: 700
+   :name: ptm-flow
+
+   PTM-enabling workflow in post-launched VM
+
+As shown in :numref:`ptm-flow`, PTM is enabled in the root port during the
+hypervisor startup. Device model then checks whether the pass through device
+supports PTM requestor capabilities and whether the corresponding root port
+supports PTM root capabilities, as well as some other sanity checks.  If an
+error is detected during these checks, the error will be reported and ACRN will
+not enable PTM on guest. This doesn’t prevent the user from launching the Guest
+VM and passing through the device to the Guest VM.  If no error is detected,
+device model will use ``add_vdev`` hypercall to add a virtual root port (VRP),
+acting as the PTM root, to Guest before passing through the device to Guest.
+
+.. figure:: images/PTM-hld-PTM-passthru.png
+   :align: center
+   :width: 700
+   :name: ptm-vrp
+
+   PTM-enabled PCI device pass-through to post-launched VM
+
+:numref:`ptm-vrp` show that, after enabling PTM, the passthru device connects to
+the virtual root port instead of the virtual host bridge.
+
+To use PTM in a virtualized environment, you may want to first verify that PTM
+is supported by the device and is enabled on the bare metal machine.
+If supported, follow these steps to enable PTM on post-launched guest:
+
+1. Make sure that PTM is enabled in guest kernel.  In Linux kernel, for example,
+   set “CONFIG_PTM=on”.
+2. Not every PCI device supports PTM.  One example that does it the Intel I225-V
+   Ethernet controller.  If you passthru this card to guest,  make sure guest
+   uses a version of the IGC driver that supports PTM.
+3. In the device model launch script, add the ``enable_ptm`` option on the
+   passthru device.  For example:
+
+   .. code-block:: bash
+      :emphasize-lines: 5
+
+      $ acrn-dm -A -m $mem_size -s 0:0,hostbridge \
+          -s 3,virtio-blk,uos-test.img \
+          -s 4,virtio-net,tap0 \
+          -s 5,virtio-console,@stdio:stdio_port \
+          -s 6,passthru,a9/00/0,enable_ptm \
+          --ovmf /usr/share/acrn/bios/OVMF.fd \
+
+4. You can check that PTM is correctly enabled on guest by displaying the PCI
+   bus hiearchy on the guest using the ``lspci`` command:
+
+   .. code-block:: bash
+      :emphasize-lines: 12,20
+
+      $ sudo lspci -tv
+        -[0000:00]-+-00.0  Network Appliance Corporation Device 1275
+         +-03.0  Red Hat, Inc. Virtio block device
+         +-04.0  Red Hat, Inc. Virtio network device
+         +-05.0  Red Hat, Inc. Virtio console
+         \-06.0-[01]----00.0  Intel Corporation Device 15f2
```

Comment:
I am confused here, as "-s 6,passthru,a9/00/0,enable_ptm \" in launch script, why 00:06.0 is assigned to vrp in guest rather than to the passthru device? It seems the passthru device is always assigned with bdf 01:00.0, which is hardcoded in acrn. 
Why not dynamically assign bdf for passthru device decided by dm option "-s <slot>" (aka. by user) and hardcode bdf for vrp?

Reply:
In order to enable PTM for guest, we need to add a virtual root port, and this virtual root port uses the passthru device's bdf (it doesn't have to but I think it is better than a random assigned bdf).  And now passthru device is connected to virtual root (a different bus) instead of host bridge and that is why the passthru device gets 01:00.0 (on bus 1 now, instead of bus 0).  Why it is hardcoded, mostly because this is what we can support in the timeframe of release 2.5, i.e., just 1 pass thru device with PTM.  In the future, for example, when there are more PTM passthru devices, we need to change this (at least the "bus" part of device's bdf).